pageserver/src/tenant/timeline.rs

```diff
@@ -3006,34 +2998,48 @@ impl Timeline {
     }
 
     // Write out the given frozen in-memory layer as a new L0 delta file
-    fn create_delta_layer(
+    async fn create_delta_layer(
         self: &Arc<Self>,
         frozen_layer: &InMemoryLayer,
     ) -> anyhow::Result<(LayerFileName, LayerFileMetadata)> {
-        // Write it out
-        let new_delta = frozen_layer.write_to_disk()?;
-        let new_delta_path = new_delta.path();
-        let new_delta_filename = new_delta.filename();
+        // TODO figure out how to use spawn_blocking. Can't use it because frozen_layer is not 'static
```

Comment:
One function up the stack (flush_frozen_layer) it is an `Arc<InMemoryLayer>` so you can clone it the usual way

Reply:
a4c4d967d76aab1413e3e07a32b149f8d4e2f0e8